### PR TITLE
Generalize Reporting & Introduce suite-by-suite HTML Reporting

### PR DESCRIPTION
**Pull request by Dzol**
This introduces and Event Manager similar to [ExUnit's](http://blog.plataformatec.com.br/2016/11/replacing-genevent-by-a-supervisor-genserver/).

We can send test information to **more than one** outputter and keep the outputter processes spinning **between** test suites. This way the outputters, namely the `HTML` outputter, can report on a **suite-by-suite** basis. The `Console` outputter is adapted (this PR makes it responsible for more of the console output we see).

### CODE REVIEW

**Comment by Dzol** [comment]
I'll resolve the failed builds on Travis: not sure why only some fail.

**Comment by Dzol** [comment]
Okay: the release notes for 1.2 write the following under enhancements.

> [GenServer] Add GenServer.stop/1 for shutting down servers reliably

File: lib/white_bread/event_manager.ex [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    import Supervisor.Spec
+    child = worker(GenServer, [], restart: :temporary)
+    Supervisor.start_link([child], strategy: :simple_one_for_one, name: __MODULE__)
+  end
+
+  def stop() do
+    for {_, pid, _, _} <- Supervisor.which_children(__MODULE__) do
+      GenServer.stop(pid, :normal, @timeout)
+    end
+    Supervisor.stop(__MODULE__)
```
Comment (mentels) [inline lib/white_bread/event_manager.ex]: As you're using `Supervisor.stop/1` there's no need to individually stop the children. As they're linked they will be stopped along with the supervisor.

**Comment by meadsteve** [comment]
Bit late looking at this but I like the approach. Thanks for the contribution :+1:  I'll merge and tag this morning.

**Comment by Dzol** [comment]
@meadsteve: Thanks for updating the README.

**Pull request merged**